lib/src/generateCPU.cc

```diff
@@ -307,6 +307,35 @@ void genNeuronFunction(const NNmodel &model, //!< Model description
                         }
                     }
 
+                    if (n.second.isInjected()) {
+                        os << "// add injected current from sources" << std::endl;
+                        os << "scalar Iinj = 0;" << std::endl;
+                        const auto scs = n.second.getCurrentSources();
+                        for (const auto *sc : scs)
+                        {
+                            const auto* scm = sc->getCurrentSourceModel();
+                            VarNameIterCtx scVars(scm->getVars());
+                            DerivedParamNameIterCtx scDerivedParams(scm->getDerivedParams());
+                            ExtraGlobalParamNameIterCtx scExtraGlobalParams(scm->getExtraGlobalParams());
+                            os << "// current source " << sc->getName() << std::endl;
+                            if (!scm->getTimeConditionCode().empty())
+                            {
+                                string tcCode = scm->getTimeConditionCode();
+                                StandardSubstitutions::currentSourceTimeCondition(tcCode, sc,
+                                                    scDerivedParams, scExtraGlobalParams);
+                                os << "if (" << tcCode << ")" << std::endl <<"{" << std::endl;
```

Comment:
As I said earlier, I don't think having a separate time condition is a good idea, but for future reference you don't need to manually insert { and }. You have two options both of which :

Automatic scopes which lets you reflect {}s in the code generator in the generated code:
```c++
{
    CodeStream::Scope b(os);
    os << "magical indented code" << std::endl;
}
```

Slightly less automatic open and closed brackets (for when you have conditionals):
```c++
os << CodeStream::OB(14);
os << "magical indented code" << std::endl
os << CodeStream::CB(14);
```

However, it probably is good to put a ``CodeStream::Scope`` inside the ``for (const auto *sc : scs)`` loop, to prevent variables from different current sources conflicting.